sample-app/src/main/java/com/lavanya/aerogearlibrary/MyApolloClient.kt

```diff
@@ -1,32 +1,83 @@
+import com.apollographql.apollo.cache.normalized.lru.LruNormalizedCacheFactory
+import com.apollographql.apollo.cache.normalized.sql.ApolloSqlHelper
+import com.apollographql.apollo.cache.normalized.sql.SqlNormalizedCacheFactory
 import okhttp3.OkHttpClient
 import okhttp3.logging.HttpLoggingInterceptor
 
-class MyApolloClient {
+class MyApolloClient : Application() {
+
+    //Implemented normalised disk cache
 
     companion object {
 
         //BASE_URL is of the format http://your_ip_adress:4000/graphql
-        private val BASE_URL = "http://192.168.0.105:4000/graphql"
-
+        private val BASE_URL = "http://192.168.43.243:4000/graphql"
         private var myApolloClient: ApolloClient? = null
+        private val SQL_CACHE_NAME = "tasksdb"
 
         fun getMyApolloClient(): ApolloClient? {
-            val loggingInterceptor = HttpLoggingInterceptor()
-            loggingInterceptor.level = HttpLoggingInterceptor.Level.BODY
+            return myApolloClient
+        }
+    }
 
-            val okHttpClient = OkHttpClient.Builder()
-                .addInterceptor(loggingInterceptor)
-                .build()
+    override fun onCreate() {
+        super.onCreate()
+        val loggingInterceptor = HttpLoggingInterceptor()
+        loggingInterceptor.level = HttpLoggingInterceptor.Level.BODY
 
-            myApolloClient = ApolloClient.builder()
-                .serverUrl(BASE_URL)
-                .okHttpClient(okHttpClient)
-                .build()
+        val okHttpClient = OkHttpClient.Builder()
+            .addInterceptor(loggingInterceptor)
+            .build()
 
-            return myApolloClient
+        val apolloSqlHelper = ApolloSqlHelper(this, SQL_CACHE_NAME)
+
+        val cacheFactory = LruNormalizedCacheFactory(
+            EvictionPolicy.NO_EVICTION, SqlNormalizedCacheFactory(apolloSqlHelper)
+        )
+
+        val resolver = object : CacheKeyResolver() {
+
+            override fun fromFieldRecordSet(field: Field, recordSet: MutableMap<String, Any>): CacheKey {
+
+                Log.e("Application", "fromFieldRecordSet ${formatCacheKey(recordSet["id"] as String)}")
+                return formatCacheKey(recordSet["id"] as String)
+
+            }
+
+            // Use this resolver to customize the key for fields with variables: eg entry(repoFullName: $repoFullName).
+            // This is useful if you want to make query to be able to resolved, even if it has never been run before.
+            override fun fromFieldArguments(field: Field, variables: Operation.Variables): CacheKey {
+
+                Log.e("Application", "fromFieldArguments ${formatCacheKey(variables.toString())}")
+
+                return CacheKey.NO_KEY
+            }
+
+            private fun formatCacheKey(id: String?): CacheKey {
+                return if (id == null || id.isEmpty()) {
+                    CacheKey.NO_KEY
+                } else {
+                    CacheKey.from(id)
```

Comment:
This will start failing when more types will be introduced. Best to just drop typename as well.

Reply:
https://github.com/apollographql/apollo-android/blob/master/apollo-sample/src/main/java/com/apollographql/apollo/sample/GitHuntApplication.java#L51-L52